lib/eslint.js

```diff
@@ -0,0 +1,259 @@
+#!/usr/src/app/bin/node_gc
+
+const fs = require("fs");
+const glob = require("glob");
+
+const eslint = require('./eslint-patch')();
+const docs = require('./docs')();
+const BatchSanitizer = require("./batch_sanitizer");
+const checks = require("./checks");
+const validateConfig = require("./validate_config");
+const computeFingerprint = require("./compute_fingerprint");
+const ConfigUpgrader = require("./config_upgrader");
+
+const CLIEngine = eslint.CLIEngine;
+const options = { extensions: [".js"], ignore: true, reset: false, useEslintrc: true };
+
+function run(runOptions) {
+  var configPath = runOptions.configPath || "/config.json";
+  var codeDir = runOptions.dir || "/code";
+
+  var cli; // instantiation delayed until after options are (potentially) modified
+  var debug = false;
+  var ignoreWarnings = false;
+  var ESLINT_WARNING_SEVERITY = 1;
+
+  var output = {
+    stdout: [],
```

Comment:
Why not send the `process` object (or, maybe `process.stdout`, `process.stderr`, and `process.exit`) as constructor arguments or as options? Then it'd be easy to stub out dummies for each.